src/gsAssembler/gsQuadrature.h

```diff
@@ -0,0 +1,82 @@
+/** @file gsQuadrature.h
+
+    @brief Creates a variety of quadrature rules
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): A. Mantzaflaris
+*/
+
+#pragma once
+
+#include <gsAssembler/gsGaussRule.h>
+#include <gsAssembler/gsLobattoRule.h>
+
+namespace gismo
+{
+
+/// Helper class for obtaining a quadrature rule
+struct gsQuadrature
+{
+    /// Quadrature rule types
+    enum rule
+    {
+        GaussLegendre = 1, ///< Gauss-Legendre quadrature
+        GaussLobatto  = 2  ///< Gauss-Lobatto quadrature
+    };
+
+    /// Constructs a quadrature rule based on input \a options
+    template<class T>
+    static gsQuadRule<T> get(const gsBasis<T> & basis,
+                             const gsOptionList & options, int fixDir = -1)
+    {
+        const int qu = options.askInt("quRule", GaussLegendre);
+        const T       quA = options.getReal("quA");
+        const index_t quB = options.getInt ("quB");
+        const gsVector<index_t> nnodes = numNodes(basis,quA,quB,fixDir);
+        return get<T>(qu, nnodes);
+    }
+
+    /// Constructs a quadrature rule based on input \a options
+    template<class T>
+    static inline gsQuadRule<T> get(int qu, gsVector<index_t> const & numNodes)
+    {
+        switch (qu)
+        {
+        case GaussLobatto :
+            return gsLobattoRule<T>(numNodes);
+        default://GaussLegendre
+            return gsGaussRule<T>(numNodes);
+        };
+    }
+
+    /// Computes and integer quA*deg_i + quB where deg_i is the degree
+    /// of \a basis
+    template<class T>
+    static gsVector<index_t> numNodes(const gsBasis<T> & basis,
+                               const T quA, const int quB, int fixDir = -1)
+    {
```

Comment:
why not index_t?

many followups...

Reply:
I want to make sure that -1 is a valid value (currently index_t=int but we can imagine index_t=unsigned)